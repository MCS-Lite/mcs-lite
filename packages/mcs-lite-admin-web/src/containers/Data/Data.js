import React from 'react';
import PropTypes from 'prop-types';
import * as R from 'ramda';
import { Observable } from 'rxjs/Observable';
import Helmet from 'react-helmet';
import A from 'mcs-lite-ui/lib/A';
import P from 'mcs-lite-ui/lib/P';
import DashboardTitle from '../../components/DashboardTitle';
import DashboardDesc from '../../components/DashboardDesc';
import DialogConfirm from '../../components/DialogConfirm';
import {
  componentFromStream,
  createEventHandler,
} from '../../utils/recomposeHelper';

const Data = componentFromStream(props$ => {
  const { handler: onResetClick, stream: onResetClick$ } = createEventHandler();
  const { handler: onCancel, stream: onCancel$ } = createEventHandler();
  const { handler: onSubmit, stream: onSubmit$ } = createEventHandler();

  const isDialogShow$ = Observable.merge(
    onResetClick$.mapTo(true),
    onCancel$.mapTo(false),
    onSubmit$.mapTo(false),
  ).startWith(false);

  // Remind: delete Side-effects.
  onSubmit$
    .withLatestFrom(props$, (e, { deleteData, getMessages: t }) =>
      deleteData.bind(null, t('reset.success')),
    )
    .subscribe(R.call);

  return props$.combineLatest(
    isDialogShow$,
    ({ getMessages: t }, isDialogShow) =>
      <div>
        <Helmet><title>{t('dataManagement')}</title></Helmet>
        <DialogConfirm
          show={isDialogShow}
          onCancel={onCancel}
          onSubmit={onSubmit}
        >
          <P>{t('reset.confirm1')}</P>
          <P>{t('reset.confirm2')}</P>
        </DialogConfirm>

        <DashboardTitle title={t('dataManagement')}>
          <A onClick={onResetClick}>{t('reset')}</A>
        </DashboardTitle>
        <DashboardDesc>{t('description')}</DashboardDesc>
      </div>,
  );
});

Data.displayName = 'Data';
Data.propTypes = {
  // Redux Action
  deleteData: PropTypes.func.isRequired,

  // React-intl I18n
  getMessages: PropTypes.func.isRequired,
};

export default Data;
