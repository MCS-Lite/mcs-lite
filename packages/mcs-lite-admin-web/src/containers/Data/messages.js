import { defineMessages } from 'react-intl';

const messages = defineMessages({
  'Data.dataManagement': {
    id: 'Data.dataManagement',
    description: 'Meta/Title',
    defaultMessage: '資料管理',
  },
  'Data.description': {
    id: 'Data.description',
    defaultMessage: '可以清除 MCS Lite 所有資料，包含使用者清單以及測試裝置等等。',
  },
  'Data.reset': {
    id: 'Data.reset',
    defaultMessage: '清除所有資料',
  },
  'Data.reset.confirm1': {
    id: 'Data.reset.confirm1',
    defaultMessage: '確定要清除所有資料嗎？',
  },
  'Data.reset.confirm2': {
    id: 'Data.reset.confirm2',
    defaultMessage: '清除後，系統將自動關閉， 請重新啟動服務。',
  },
  'Data.reset.success': {
    id: 'Data.reset.success',
    defaultMessage: '成功！已清除所有資料。',
  },
});

export default messages;
