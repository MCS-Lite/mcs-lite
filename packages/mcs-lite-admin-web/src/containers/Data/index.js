import { connect } from 'react-redux';
import compose from 'recompose/compose';
import { withGetMessages } from 'react-intl-inject-hoc';
import messages from './messages';
import { actions } from '../../modules/data';
import Data from './Data';

export const mapDispatchToProps = { deleteData: actions.deleteData };

export default compose(
  connect(null, mapDispatchToProps),
  withGetMessages(messages, 'Data'),
)(Data);
