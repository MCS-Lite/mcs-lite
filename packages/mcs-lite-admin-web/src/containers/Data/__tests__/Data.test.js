import React from 'react';
import R from 'ramda';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Data from '../Data';

it('should renders <Data> correctly without ip', () => {
  const wrapper = shallow(
    <Data getMessages={R.identity} deleteData={() => {}} />,
  );

  expect(toJson(wrapper)).toMatchSnapshot();
});
