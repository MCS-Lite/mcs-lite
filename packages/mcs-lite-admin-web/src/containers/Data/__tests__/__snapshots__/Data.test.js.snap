// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should renders <Data> correctly without ip 1`] = `
<div>
  <HelmetWrapper
    encodeSpecialCharacters={true}
  >
    <title>
      dataManagement
    </title>
  </HelmetWrapper>
  <InjectIntl(withPropsOnChange(DialogConfirm))
    onCancel={[Function]}
    onSubmit={[Function]}
    show={false}
  >
    <P>
      reset.confirm1
    </P>
    <P>
      reset.confirm2
    </P>
  </InjectIntl(withPropsOnChange(DialogConfirm))>
  <DashboardTitle
    title="dataManagement"
  >
    <A
      color="primary"
      onClick={[Function]}
    >
      reset
    </A>
  </DashboardTitle>
  <DashboardDesc>
    description
  </DashboardDesc>
</div>
`;
